File: src/angular/hq/src/app/models/Invoices/get-invoices-v1.ts
import { Period } from '../../enums/period';
import { PagedRequestV1 } from '../common/paged-request-v1';
import { PagedResponseV1 } from '../common/paged-response-v1';
import { SortDirection } from '../common/sort-direction';

export interface GetInvoicesRequestV1 extends PagedRequestV1 {
  search?: string | null;
  id?: string;
  clientId?: string;
  sortBy: SortColumn;
  sortDirection: SortDirection;
  period?: Period | null;
  startDate?: Date | null;
  endDate?: Date | null;
}

export enum SortColumn {
  ClientName = 1,
  InvoiceNumber = 2,
  Total = 3,
  TotalApprovedHours = 4,
  Date = 5,
}

export interface GetInvoicesRecordV1 {
  id: string;
  clientId: string;
  clientName: string;
  date: Date;
  invoiceNumber: string;
  total: number;
  totalApprovedHours: number;
}

export interface GetInvoicesRecordsV1 {
  records: [GetInvoicesRecordV1];
  total: number | null;
}

export interface GetInvoicesResponseV1
  extends PagedResponseV1<GetInvoicesRecordV1> {}

File: src/dotnet/HQ.Abstractions/Invoices/GetInvoicesV1.cs


using HQ.Abstractions.Common;
using HQ.Abstractions.Enumerations;


namespace HQ.Abstractions.Invoices
{
    public class GetInvoicesV1
    {
        public class Request : PagedRequestV1
        {
            public string? Search { get; set; }
            public Guid? Id { get; set; }
            public Guid? clientId { get; set; }
            public SortColumn SortBy { get; set; } = SortColumn.ClientName;
            public SortDirection SortDirection { get; set; } = SortDirection.Asc;
        }
        public enum SortColumn
        {
            ClientName = 1,
            InvoiceNumber = 2,
            Total = 3,
            TotalApprovedHours = 4,
            Date = 5,
        }

        public class Response : PagedResponseV1<Record>;
        public class Record
        {
            public Guid Id { get; set; }
            public Guid ClientId { get; set; }
            public required string ClientName { get; set; }
            public DateOnly Date { get; set; }
            public string? InvoiceNumber { get; set; }
            public decimal Total { get; set; }
            public decimal TotalApprovedHours { get; set; }
            public DateOnly CreatedAt { get; set; }
        }
    }
}